Content.Server/Morgue/MorgueSystem.Crematorium.cs

```diff
@@ -0,0 +1,134 @@
+using Content.Server.Morgue.Components;
+using Content.Shared.Morgue;
+using Robust.Server.GameObjects;
+using Robust.Shared.Player;
+using Robust.Shared.Audio;
+using Content.Server.Storage.Components;
+using System.Threading;
+using Content.Shared.Verbs;
+using Content.Shared.Database;
+using Content.Shared.Interaction.Events;
+using Content.Server.Players;
+
+namespace Content.Server.Morgue;
+
+/// <summary>
+///    This is all the crematorium specific logic
+/// </summary>
+public sealed partial class MorgueSystem : EntitySystem
+{
+    private void AddCremateVerb(EntityUid uid, MorgueComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!component.IsCrematorium)
+            return;
+
+        if (!TryComp<EntityStorageComponent>(component.Tray, out var storage))
+            return;
+
+        if (!args.CanAccess || !args.CanInteract || args.Hands == null || component.Cooking || storage.Open)
+            return;
+
+        AlternativeVerb verb = new()
+        {
+            Text = Loc.GetString("cremate-verb-get-data-text"),
+            // TODO VERB ICON add flame/burn symbol?
+            Act = () => TryCremate(uid, component, storage),
+            Impact = LogImpact.Medium // could be a body? or evidence? I dunno.
+        };
+        args.Verbs.Add(verb);
+    }
+
+    public void Cremate(EntityUid uid, MorgueComponent? component = null, EntityStorageComponent? storage = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        if (!component.IsCrematorium)
+            return;
```

Comment:
morgue and crematorium should probably just be made separate components. its not clear to my why having them be the same component helps